src/core/DY.Core.Trace.Manipulation.fst

```diff
@@ -254,8 +255,8 @@ val mk_rand_trace_invariant:
   (ensures (
     let (b, tr_out) = mk_rand usg lab len tr in
     trace_invariant tr_out /\
-    1 <= DY.Core.Trace.Type.length tr_out /\
-    rand_generated_at tr_out (DY.Core.Trace.Type.length tr_out - 1) b
+    1 <= DY.Core.Trace.Base.length tr_out /\
```

Comment:
See previous comment about wrapping this up in a predicate --- strengthened by the fact that this use case is `1 <= length` while the other is `0 < length`, which should be equivalent (but I've seen F* fail to nicely convert between these in other settings).

Reply:
Ah correct, maybe we could define something like `val rand_just_generated: trace -> bytes -> prop` (or some better name) in a separate PR

Reply:
Moving the discussion in #61.